### ISSUE TITLE
CP can be automatically downloaded to your PC

### ISSUE THREAD

**Issue by Guedez** [HelloZeroNet/ZeroNet#2133]
If a troll starts spamming a chat that enables images with child pornography, your computer will download and host those images, no?

Could it be possible to enable some fractional image mode? Where your computer never saves a whole image, so no matter what's posted, you never host a whole image, so you are never "hosting child porn".

This is the only reason I am not using this yet, some countries have some pretty draconian laws about that

**Comment by April93** [comment]
>If a troll starts spamming a chat, that enables images, with child pornography, your computer will download and host those images, no?

Yes and no. Yes if the site includes images that are included as a regular post or if you are viewing the site and it doesn't have a "click to view" type system. But if the site is structured like zerome, where images are optional files, and you don't auto-view them without clicking on them, your computer will not automatically download them unless you click on them. It's a per-site thing, so basically you just want to be careful what you click on.

>Could it be possible to enable some fractional image mode? Where your computer never saves a whole image, so no matter what's posted, you never host a whole image, so you are never "hosting child porn".

AFAIK this isn't possible with the zeronet framework. When you post something you're the only one originally hosting it. And then others fetch it from you when they view it. When you view an image, it downloads the entire image, just as other forms of the web do. As you can't actually display something on your computer without downloading it. Normally this is done in the background and is called a "cache". What you're talking about is something other p2p webs do, where they divide up a file between multiple "always online" users, and then reconstruct it in the cache upon viewing, and then removing it when stopped looking at it. ZeroNet isn't built to handle files in this way, and doing that would remove the ability for zeronet to work seamlessly offline+online+meshes.

Simply running zeronet itself *will not* download anything you do not wish to see. ZeroNet does not automatically download anything. If you visit a site, it will download the site, regardless of what's on it. Sites can have "optional files" which aren't downloaded until you request the file (by visiting the url or having a page fetch that file/url). You can delete and/or stop seeding both sites and their optional files.

So some examples of sites with images:

ZeroMe: Has image uploading features. When you visit zerome you will not have any content on it. You then download hubs. Hubs will have text posts and usually have optional image files. The optional files are only loaded when you directly visit the link, or click on a blurred image on the site. It will then download the image.

0chan: I'm not sure if 0chan uses optional files, but from my experience the chans don't have the same "click to view" feature that zeronet does. Meaning that if you load one of the pages with an image, it downloads the image automatically. If they aren't using optional files, if you have the site downloaded, it will automatically fetch new images uploaded to the site.

So you have:

Site with no base images, but click-to-view optional files: No problem with these. You can browse the site freely and just avoid clicking on unknown or known pedophilic images. If you do click on one, you can delete it and move on.

Sites with no base images, but optional files that load as you see them: Slight problem as you auto-download the images you end up seeing. So you know if you downloaded something not so good like cp. You can delete the optional files and move on.

Sites with base images that are not optional: These are usually blogs. Simply having the site downloaded will auto-fetch any new content posted on it, including the images. Even if you didn't yet look at the content. Deleting the site will delete the content. IIRC the chans may fall into this category. or the one above?

**Comment by Guedez** [comment]
Hmmm, I understand why they can't be partially stored
Is there an easy way to delete only images from the cache? Like the browsers have a "clear cache" button, maybe zeronet could have a "delete images and don't download those again" to quickly get rid of those unwanted images.
For instance, if you go to a chan board catalog, you will be download a whole lot of images that you don't see if you don't go through the effort of actually checking all of them manually. If I could just purge all images from my cache easily, would solve a lot of problems. Especially if this could be somehow automated. Maybe set some sites as "delete all 1-day old images"

**Comment by skwerlman** [comment]
0chan has thumbnails as regular files, but the full-res versions are optional files

**Comment by imachug** [comment]
This sounds like a user-controlled optional file list, as opposed to site-owner-controlled list. Am I understanding the proposal right?

**Comment by Guedez** [comment]
At first I was asking to be able to only partially host a image, like torrent files are hosted in pieces. But it was explained that it's not possible.
But now, #2138 probably solves all issues, as it's easy to quickly delete a image if you really don't want it on your pc, even on your cache, and you can even disable file hosting, so you can be pretty much safe, since it becomes just as dangerous as normal sites that might download things to your cache. If you are never hosting files, then it's the same as any other site

**Comment by HostFat** [comment]
On 08chan and Millchan, before opening any boards, you can click on "options" on the upper left, and set "images preview: 0"
On this way, if you open a board, still images will not be pre-loaded.

Other then this, I had the same your concern about cp images.

So without opening/clicking any boards that may contain cp images, I've looked to the data folder of the website, and the folder board that may contain those images.

The folder had only the "content.json" and "data.json", no images. ( as I didn't open it)

**Comment by April93** [comment]
> On 08chan and Millchan, before opening any boards, you can click on "options" on the upper left, and set "images preview: 0"
> On this way, if you open a board, still images will not be pre-loaded.

I know on millchan thumbnails are optional files. But is the same true for 08chan? Last time I checked this wasn't true.

> So without opening/clicking any boards that may contain cp images, I've looked to the data folder of the website, and the folder board that may contain those images.
> 
> The folder had only the "content.json" and "data.json", no images. ( as I didn't open it)
> 
> I saw instead images on other boards that I had open before.

AFAIK 08chan has thumbnails that are *not* optional files, but has the full images which *are* optional files. Thus it'd download the thumbnails even when you don't look at them. This is why I avoid using the site. There's another site, the [millchan merger site](http://127.0.0.1:43110/17gCoQDCAAoQkzKRiajFNUAhnVD2gSNmuC/), which resolves this problem.

**Comment by HostFat** [comment]
> This sounds like a user-controlled optional file list, as opposed to site-owner-controlled list. Am I understanding the proposal right?

A general setting and a x site setting, so that Zeronet will NOT download any image from a site (only text), that the user can enable/change **before** downloading anything, **can be very good.**

If the user enable it (for every sites or just some), then the user should also be able to downloading them (the images), by clicking on them one by one.
